frontend/app/src/components/home/SigninWithSol.svelte

```diff
@@ -0,0 +1,155 @@
+        (wallet) => wallet.readyState === "Installed",
+    ).length;
+
+    function walletError(error: WalletError): void {
+        console.error("WalletError: ", error);
+    }
+
+    async function connectWallet(name: WalletName) {
+        try {
+            connecting = name;
+            await select(name);
+            await connect();
+            if (publicKey && wallet && signMessage) {
+                const account = publicKey.toBase58();
+                const prepareResponse = await client.siwsPrepareLogin(account);
+
+                if (prepareResponse.kind === "success") {
+                    const request = buildSignInRequest(prepareResponse.siwsMessage);
+                    const data = new TextEncoder().encode(request);
+                    const signResponse = await signMessage(data);
+                    const signature = base58.encode(signResponse);
+                    await client.signInWithWallet("sol", account, signature);
+                }
+            } else {
+                console.error("Didn't get an address back from the connector");
+            }
+        } catch (err) {
+            console.error(`Error connecting to wallet: ${name}`, err);
+        } finally {
+            connecting = undefined;
+        }
+    }
+
+    onMount(async () => {
+        const connection = new Connection(clusterApiUrl("mainnet-beta"), "processed");
+        console.log("Connection: ", connection);
+        initialize({
+            wallets: [
+                new PhantomWalletAdapter(),
+                new WalletConnectWalletAdapter({
+                    network: WalletAdapterNetwork.Mainnet,
+                    options: {
+                        projectId: process.env.WALLET_CONNECT_PROJECT_ID!,
+                    },
+                }),
+            ],
+            autoConnect: true,
+            localStorageKey,
+            onError: walletError,
+        });
+    });
+
+    function buildSignInRequest(siwsMessage: SiwsMessage): string {
+        // expiration_time and issued_at are in nanoseconds, convert to milliseconds.
+        const expMilliseconds = Number(
+            siwsMessage.expirationTime / BigInt(1000000)
+        );
+        const issuedAtMilliseconds = Number(
+            siwsMessage.issuedAt / BigInt(1000000)
+        );
+
+        let request = `${siwsMessage.domain} wants you to sign in with your Solana account:\n${siwsMessage.address}\n\n`;
+        request += `${siwsMessage.statement}\n\n`;
+        request += `URI: ${siwsMessage.uri}\n`;
+        request += `Version: ${siwsMessage.version}\n`;
+        request += `Chain ID: ${siwsMessage.chainId}\n`;
+        request += `Nonce: ${siwsMessage.nonce}\n`;
+        request += `Issued At: ${new Date(issuedAtMilliseconds).toISOString()}\n`;
+        request += `Expiration Time: ${new Date(expMilliseconds).toISOString()}`;
+        return request;
+    }
+</script>
+
+<h1>
+    {walletsAvailable
+        ? "Connect a wallet to continue"
```

Comment:
These should have translations.

Reply:
These have been removed now, so it matches sign in with ETH